skills/nemo-relay-migrate-from-flow/scripts/migrate_from_nemo_flow.py

```
@@ -540,16 +637,24 @@ def main() -> int:
+        legacy_configs = collect_legacy_project_configs(root, args.include_generated)
+
         file_changes = [
             change
             for path in iter_files(root, args.include_lockfiles, args.include_generated)
+            if not is_legacy_project_config(path)
             if (change := rewrite_file(path, root, args.write, root_fd)) is not None
         ]
 
+        current_legacy_configs = collect_legacy_project_configs(root, args.include_generated)
+        if args.write and legacy_configs_changed(legacy_configs, current_legacy_configs):
+            raise MutationError("legacy project configuration changed during scan")
```

Ruff (0.16.1)

[warning] Abstract `raise` to an inner function

(TRY301)